sentry_sdk/profiler.py

```diff
@@ -0,0 +1,186 @@
+"""
+This file is originally based on code from https://github.com/nylas/nylas-perftools, which is published under the following license:
+
+The MIT License (MIT)
+
+Copyright (c) 2014 Nylas
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import atexit
+from datetime import datetime
+import signal
+import time
+import threading
+import json
+from sentry_sdk.utils import logger
+
+# TODO: temp import to output file
+import os
+
+def nanosecond_time():
+    return int(time.perf_counter() * 1e9)
+
+class FrameData:
+    def __init__(self, frame):
+        self.function_name = frame.f_code.co_name
+        self.module = frame.f_globals['__name__']
+
+        # Depending on Python version, frame.f_code.co_filename either stores just the file name or the entire absolute path.
+        self.file_name = frame.f_code.co_filename
+        self.line_number = frame.f_code.co_firstlineno
+    
+    @property
+    def _attribute_tuple(self):
+        """Returns a tuple of the attributes used in comparison"""
+        return (self.function_name, self.module, self.file_name, self.line_number)
+
+    def __eq__(self, other):
+        if isinstance(other, FrameData):
+            return self._attribute_tuple == other._attribute_tuple
+        return False
+    
+    def __hash__(self):
+        return hash(self._attribute_tuple)
+
+    def __str__(self):
+        return f'{self.function_name}({self.module}) in {self.file_name}:{self.line_number}'
```

Comment:
our code needs to run in python2.7 too (or at least needs to be parsed for CI to work)
so please use old style formatted strings and not f-strings

Reply:
Sounds good! I decided to remove these `__str__` methods because I do not think they are being called from anywhere (I had used them for testing/debugging purposes earlier). But, there was one other instance of an f-string in this file that I have fixed.